primer/brand
# Apply text-decoration none to :hover state for Button and InlineLink

## Summary

Components that may also serve as links (such as [`Button`](https://github.com/primer/brand/blob/6d590e2994398caceee5731a2f59e13d108f2b9e/packages/react/src/Button/Button.module.css#L8) or [`InlineLink`](https://github.com/primer/brand/blob/6d590e2994398caceee5731a2f59e13d108f2b9e/packages/react/src/InlineLink/InlineLink.module.css#L2)) have `text-decoration:none` rules only applied to their normal state. This can create conflicts when sites use a stylesheet that explicitly sets `text-decoration:underline` for `a:hover` (for example, for accessibility reasons). 

I think adding `text-decoration:none` to the `:hover` state would fix the problem.

![Screenshot 2023-04-25 at 16 32 15](https://user-images.githubusercontent.com/3042654/234310912-c97f7f40-d9d6-4782-8838-2eb553936d5d.png)


## List of notable changes:

- **updated** the Button and InlineLink element to explicitly set the text-decoration: underline property to none on the :hover state


## Steps to test:

1. Open the Button component in CI-deployed preview environment for Storybook
2. Verify that Button applies the text-decoration none CSS property explicitly on the :hover state
3. Open the InlineLink component in CI-deployed preview environment for Storybook
4. Verify that InlineLink applies the text-decoration none CSS property explicitly on the :hover state

## Supporting resources (related issues, external links, etc):

- https://github.com/primer/brand/issues/247

## Contributor checklist:


## Reviewer checklist:
